src/common/io.py

```diff
@@ -1807,13 +1813,22 @@ def integration_point(self, solver = None):
         max_size = self.options.get("result_max_size", None)
         if max_size is not None:
             current_size = sys.getsizeof(self.time_sol) + sys.getsizeof(self.real_sol) + \
-                            sys.getsizeof(self.int_sol)  + sys.getsizeof(self.bool_sol) + \
-                            sys.getsizeof(self.param_sol)
+                           sys.getsizeof(self.int_sol)  + sys.getsizeof(self.bool_sol) + \
+                           sys.getsizeof(self.param_sol)
+            if self._first_point:
```

Comment:
Except for the "how" of determining the sizes, the code appears identical for all result handlers. I'd suggest to break it out into an independent function instead.

Additionally, I'd prefer to have the magic number of 5000 defined as e.g., "NCP_LARGE = 5000" defined somewhere in the beginning of the file.

Reply:
Fixed.